### PR TITLE
Bugfix: onUpdated does not get trigger on adding new elements

### PR DESCRIPTION
**Pull request by derrandz**
This commit is a suggestion to make `data.save` responsible for dispatching `formeoUpdated`, thus I removed `dispatchEvent(formeoUpdated)` from various places where `data.save`, incuding `removeEmpty`

### CODE REVIEW

**Comment by kevinchappell** [comment]
formUpdated was originally called from data.save but was moved to all those other locations for a reason i cant remember now. I think because until was have history stack, we have no way of knowing previous changes, and `onUpdate` will eventually have previous and current formData and contain only the changes. Still, its prob better to move back to `data.save`.

File: demo/assets/js/demo.js [hunk 1]
```diff
@@ -12,7 +12,43 @@ function readyState() {
...
       debug: true,
       sessionStorage: true,
-      editPanelOrder: ['attrs', 'options']
+      editPanelOrder: ['attrs', 'options'],
+      events: {
+        onUpdate: function (evt) {
+          console.log("Updated on the spot");
+        },
+      },
+      controls: {
```
Comment (kevinchappell) [inline demo/assets/js/demo.js]: Need to remove test config

File: src/js/common/data.js [hunk 2]
```diff
@@ -330,6 +330,8 @@ var data = {
       console.log('Saved: ' + group);
     }
 
+    // Shouldn't be the case? because everytime save is called there should be some formData update, right?
```
Comment (kevinchappell) [inline src/js/common/data.js:333]: correct, we should move back here but the `formeoUpdated` custom event detail will need to be updated as well.

**Comment by derrandz** [comment]
I suppose we need some design to clear out how to dispatch events properly, without having to manually redispatch the `onUpdate` everytime some update is made.

**Pull request merged**